src/cxx_support.rs

```diff
@@ -0,0 +1,214 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Support for cxx types.
+
+use std::pin::Pin;
+
+use cxx::{memory::UniquePtrTarget, UniquePtr};
+
+use crate::{Emplace, TryNew};
+
+/// Equivalent of a [`UniquePtr`] which is allocated, but hasn't yet
+/// had its constructor called.
+#[cfg(feature = "cxx")]
+pub struct MaybeUninitUniquePtr<T: UniquePtrTarget>(*mut T);
+
+impl<T: UniquePtrTarget> MaybeUninitUniquePtr<T> {
+  /// Declare that a pointer which we have received from C++
+  /// is, in fact, not initialized but is instead just a blank
+  /// storage area yet to have its constructor called.
+  ///
+  /// Safety: destroying this type results in a memory leak. Callers
```

Comment:
Use a new header (`# Safety`) for the safety section.

Reply:
Will do